### PR TITLE
Add several spelling corrections

### PR DESCRIPTION
**Pull request by luzpaz**
Note: some typos mentioned found in https://github.com/FreeCAD/FreeCAD/pull/15509#pullrequestreview-2200816118

### CODE REVIEW

File: codespell_lib/data/dictionary.txt [hunk 1]
```diff
@@ -14709,6 +14712,7 @@ comverter->converter
 comverters->converters
 comverting->converting
 comverts->converts
+comvolution->convolution
```
Comment (DimitriPapadopoulos) [inline codespell_lib/data/dictionary.txt:14715]: Perhaps add plural?
```suggestion
comvolution->convolution
comvolutions->convolutions
```

File: codespell_lib/data/dictionary.txt [hunk 2]
```diff
@@ -19586,6 +19590,7 @@ demagogs->demagogues
 demaind->demand
 demandin->demanding, demand in,
 demaned->demand, demeaned,
+demaond->demand
```
Comment (DimitriPapadopoulos) [inline codespell_lib/data/dictionary.txt:19593]: Add plural?
```suggestion
demaond->demand
demaonds->demands
```

File: codespell_lib/data/dictionary.txt [hunk 3]
```diff
@@ -36868,6 +36874,7 @@ mathods->methods
 matinay->matinee
 matirx->matrix
 matix->matrix
+matrces->matrices, mattresses,
```
Comment (DimitriPapadopoulos) [inline codespell_lib/data/dictionary.txt:36877]: I think `matrices` is sufficient here.
```suggestion
matrces->matrices
```

File: codespell_lib/data/dictionary.txt [hunk 4]
```diff
@@ -37701,6 +37709,7 @@ mirorred->mirrored
 mirorring->mirroring
 mirorrs->mirrors
 mirors->mirrors, minors,
+mirrir->mirror
```
Comment (DimitriPapadopoulos) [inline codespell_lib/data/dictionary.txt:37712]:
```suggestion
mirrir->mirror
mirrired->mirrored
mirriring->mirroring
mirrirs->mirrors
```

File: codespell_lib/data/dictionary.txt [hunk 5]
```diff
@@ -48018,6 +48027,7 @@ reduceable->reducible
 reducin->reducing
 redudancy->redundancy
 redudant->redundant
+redumndant->redundant
```
Comment (DimitriPapadopoulos) [inline codespell_lib/data/dictionary.txt:48030]:
```suggestion
redumndancy->redundancy
redumndant->redundant
```

**Comment by luzpaz** [comment]
Thanks @DimitriPapadopoulos. I can't seem to merge your diffs via github interface. Do you mind taking a shot at merging them  ?

**Review by DimitriPapadopoulos — APPROVED** [review]

**Comment by DimitriPapadopoulos** [comment]
Thanks @luzpaz!

**Pull request merged**